modules/git/commit.go

```diff
@@ -160,30 +160,42 @@ func AllCommitsCount(ctx context.Context, repoPath string, hidePRRefs bool, file
 	return strconv.ParseInt(strings.TrimSpace(stdout), 10, 64)
 }
 
-// CommitsCountFiles returns number of total commits of until given revision.
-func CommitsCountFiles(ctx context.Context, repoPath string, revision, relpath []string) (int64, error) {
+// CommitsCountOptions the options when counting commits
+type CommitsCountOptions struct {
+	RepoPath string
+	Not      string
+	Revision []string
+	RelPath  []string
+}
+
+// CommitsCount returns number of total commits of until given revision.
+func CommitsCount(ctx context.Context, opts CommitsCountOptions) (int64, error) {
 	cmd := NewCommand(ctx, "rev-list", "--count")
-	cmd.AddDynamicArguments(revision...)
-	if len(relpath) > 0 {
-		cmd.AddDashesAndList(relpath...)
+
+	cmd.AddDynamicArguments(opts.Revision...)
+
+	if opts.Not != "" {
+		cmd.AddOptionValues("--not", opts.Not)
```

Comment:
Sorry for bothering ... the last nit (hopefully)

It's better to put  `AddOptionValues`  before `AddDynamicArguments` to avoid breaking git command

The widely used order is:  `option-values` -> `dynamic arguments` -> `dashes + file list`

Reply:
I've found that the command does not work as expected when adding  `--not` before the revision.

For example, if I make a new branch on a repo off master, named `branch1`. Then add 1 commit

```
git rev-list --not master --count branch1
```
Always returns 0

But
```
git rev-list --count branch1 --not master
```

Returns 1 as expected

Reply:
You can even see in the documentation here, under the `Description` section, they show an example where `--not` is placed after the revision

https://git-scm.com/docs/git-rev-list

```
git rev-list A B --not $(git merge-base --all A B)
```

Reply:
Oh, you are right, thank you ~~~